client/my-sites/menus/menus-revert-button.jsx

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+
+const debug = debugFactory( 'calypso:menus:revert-button' ); // eslint-disable-line no-unused-vars
+
+const MenuRevertButton = React.createClass( {
+	componentWillMount() {
+		this.props.menuData.on( 'saved', this.setReverted );
+		this.props.menuData.on( 'error', this.setReverted );
+	},
+
+	componentWillUnmount() {
+		this.props.menuData.off( 'saved', this.setReverted );
+		this.props.menuData.off( 'error', this.setReverted );
+	},
+
+	getInitialState() {
+		return { reverting: false };
+	},
+
+	setReverting() {
```

Comment:
Good question, I hadn't considered those names. `reverting` and `reverted` are states, which is what these methods set. `start` and `end` seem more like events, which would imply that these methods would emit something. Also, their usage in the code seems more natural with `setReverting()` and `setReverted()` than start/end. Based on that reasoning I prefer their current names, but am interested in hearing arguments for start/end.